### PR TITLE
feat: allow setting window shape

### PR DESCRIPTION
**Pull request by nornagon**
This binds Widget::SetShape, an API that already exists in Chromium. It's a more reliable method of having some parts of your window be "click-through" than the current `setIgnoreMouseEvents` API, which messes around with the `WS_EX_LAYERED` window style, causing strange bugs and incompatibility with hardware acceleration.

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -1262,6 +1262,11 @@ Sets the opacity of the window. On Linux does nothing.
 
 Returns `Number` - between 0.0 (fully transparent) and 1.0 (fully opaque)
 
+#### `win.setShape(rects)` _Windows_ _Linux_ _Experimental_
+
+* `rects` [Rectangle[]](structures/rectangle.md) - Sets a shape on the window.
```
Comment (zeke) [inline docs/api/browser-window.md]: Can you expand this documentation a bit to explain what shapes are and why you'd want to use them?

Comment (nornagon) [reply docs/api/browser-window.md]: done

**Pull request merged**